### DISCUSSION TITLE
Yolo version for Instance Segmentation

### DISCUSSION THREAD

**Discussion by ghost** [ultralytics/ultralytics#8689] [Q&A]
Hello,

Which version of yolo(v5,v8) is easier to customize(change backbone,modify architecture,change mask predict network etc) for instance segmentation tasks? Also, where can i find documentation for the code and for the architecture?

Thank you

**Comment by pderrenger** [comment]
Hello!

For instance segmentation tasks, YOLOv8 is your go-to choice as it's designed with enhanced capabilities, including instance segmentation, and offers more flexibility for customization like changing the backbone, modifying the architecture, and adjusting the mask prediction network. You can find comprehensive documentation for YOLOv8, including code and architecture details, at our official documentation site: [YOLOv8 Documentation](https://docs.ultralytics.com/models/yolov8).

Here's a quick example of how you might load and customize a YOLOv8 model for instance segmentation:

```python
from ultralytics import YOLO

# Load a pretrained YOLOv8 model
model = YOLO('yolov8-seg.pt')  # For instance segmentation

# Example: Change the backbone (hypothetical code)
model.backbone = CustomBackbone()

# For more detailed customization, refer to the documentation and the codebase.
```

Feel free to explore and customize as per your needs! 😊 If you have any more questions or need further assistance, don't hesitate to ask.

Happy coding!
